source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+       auto_generate_hydrator_classes: true
+       connections:
+           default:
+               server: '%env(resolve:MONGODB_URL)%&appName=devrel.content.php'
+               options: {}
+       default_database: '%env(resolve:MONGODB_DB)%'
+       document_managers:
+           default:
+               auto_mapping: true
+               mappings:
+                   App:
+                       dir: '%kernel.project_dir%/src/Document'
+                       mapping: true
+                       type: attribute
+                       prefix: 'App\Document'
+                       is_bundle: false
+                       alias: App
+
+Install Frontend Dependency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This project uses ``twig``, the default template engine for Symfony, to
+generate templates in this application. Run the following command to
+install the ``twig`` bundle:
+
+.. code-block::
+
+   composer require symfony/twig-bundle
+
+.. note::
+
+   This step might result in an error message about unset environment
+   variables, but this is resolved in the following section.
+
+Modify Project Files
+~~~~~~~~~~~~~~~~~~~~
+
+This section demonstrates how to modify the files in your
+``restaurants`` project to create a Symfony web application that views
+restaurants based on filter criteria.
+
+Set Environment Variables
+`````````````````````````
+
+In the root directory, navigate to the ``.env`` file and define the
+following environment variables or edit them if they are already present:
+
+.. code-block:: none
+   :caption: .env
+
+   ...
+
+   MONGODB_URL=<your Atlas connection string>
+   MONGODB_DB=sample_restaurants
+
+To retrieve your connection string, see the
+:ref:`php-symfony-atlas-cluster` step.
+
+Create Restaurant Entity and Controller
+```````````````````````````````````````
+
+Create the ``Restaurant.php`` file in the ``src/Document`` directory and
+paste the following code to create an entity that represents documents in
+the ``restaurants`` collection:
+
+.. literalinclude:: /includes/php-frameworks/Restaurant.php
+   :caption: src/Document/Restaurant.php
+   :language: php
+
+Next, create the ``RestaurantController.php`` file in the
+``src/Controller`` directory and paste the following code to handle the
+endpoints in your application:
+
+.. literalinclude:: /includes/php-frameworks/RestaurantController.php
+   :caption: src/Controller/RestaurantController.php
+   :language: php
+
+The controller file defines the ``index()`` method, which displays text on
+the web app's front page. The file also defines the ``browse()`` method,
+which finds documents in which the ``borough``
+field is ``'Queens'`` and the ``name`` field contains the string ``'Moon'``.
+This method then displays the documents at the ``/restaurant/browse/`` route. The
+``browse()`` method uses the ``QueryBuilder`` class to construct the query.
+
+Specify Routes
+``````````````
+
+In the ``config/routes.yaml`` file, add the following routes:
```

Comment:
@GromNaN: Is this section even necessary since the controller methods have Route attributes? I would have assumed Symfony handles this on its own.

Reply:
Assuming this was also copied from the dev rel tutorial, I expect there's some redundancy there as well. See: https://symfony.com/doc/current/routing.html

@alcaeus: Should this section be removed entirely? Perhaps "This file sets the following routes in the application" should be relocated to the previous section where the controller class is created, since that ends up defining routes via method attributes.

Reply:
Yep, there's no need to specify routes manually. In fact, I'd highly suggest we stick to Symfony Best Practices and use `Route` attributes on controller actions to configure the router and do away with the config files.

Reply:
@rustagir: This entire section can be removed.